observ/watcher.py

```diff
@@ -9,27 +9,28 @@
 from functools import partial, wraps
 import inspect
 from itertools import count
-from typing import Any, Callable, Optional, TypeVar
+from typing import Any, Callable, Generic, TypeVar
 from weakref import ref, WeakSet
 
 from .dep import Dep
 from .dict_proxy import DictProxyBase
 from .list_proxy import ListProxyBase
-from .proxy import Proxy
 from .scheduler import scheduler
 from .set_proxy import SetProxyBase
 
 
-T = TypeVar("T", bound=Callable[[], Any])
+T = TypeVar("T")
+Watchable = Callable[[], T] | T
```

Comment:
Feels like some detail was lost here, not everything is watchable. We only support a callable, or directly a proxy, or a list of proxies.